src/JukeBox.cpp

```diff
@@ -33,46 +50,198 @@ bool JukeBox::start(string exeName, string configFilePath) {
   return true;
 }
 
-TResult<TSessionID> JukeBox::generateSession(
-    optional<TPassword> const &pw, std::optional<std::string> const &nickname) {
-  return Error(ErrorCode::NotImplemented,
-               "generateSession is not implemented yet");
+TResult<TSessionID> JukeBox::generateSession(optional<TPassword> const &pw,
+                                             optional<string> const &nickname) {
+  User user;
+  auto conf = ConfigHandler::getInstance();
+  auto adminPw = conf->getValueString("MainParams", "adminPassword");
+  if (holds_alternative<Error>(adminPw))
+    return get<Error>(adminPw);
+
+  string name = "(no nickname given)";
+  if (nickname.has_value()) {
+    name = nickname.value();
+    user.Name = name;
+  }
+
+  if (pw.has_value() && pw.value() == get<string>(adminPw)) {
+    LOG(INFO) << "JukeBox.generateSession: User '" << name << "' is admin!";
+    user.isAdmin = true;
+  }
+  mDataStore->addUser(user);
+
+  return static_cast<TSessionID>(to_string(time(nullptr)));
```

Comment:
i dont care how the session is generated but maybe you should add it to the user? :D
(btw a simple uniqueness check should be done too)

Reply:
good catch :grinning: 
i just realized that using `time()` also limits the maximum call frequency for this function to 1 Hz.
more frequent calls will generate the same ID within a second..

--> implementing a different ID generation "algorithm".